src/components/Convert/routes/hlpToBalancer.ts

```diff
@@ -0,0 +1,163 @@
+  const hlpBalancerToken = pool?.tokensInLp.find(
+    (token) =>
+      token.extensions?.isHlpToken && token.address.toLowerCase() !== to.address.toLowerCase()
+  )!;
+
+  return {
+    tokenIn: from,
+    hlpBalancerToken: hlpBalancerToken,
+    tokenOut: to,
+    pool
+  };
+};
+
+const hlpToBalancerWeight = async ({
+  fromToken,
+  toToken,
+  network
+}: WeightInput): Promise<number> => {
+  console.log();
+  if (!fromToken.extensions?.isHlpToken) return 0;
+  if (!config.protocol[network]?.protocol.routerHlpBalancer) return 0;
+
+  const { pool } = getHlpToBalancerPath(fromToken, toToken);
+
+  // the statement below should be removed if/when other types of pools are supported
+  if (pool.type !== "Weighted") return 0;
+
+  return HLP_TO_BALANCER;
+};
+
+const hlpToBalancerQuote = async (input: ConvertQuoteRouteArgs): Promise<RawQuote> => {
+  if (!input.signerOrProvider) throw new Error("signerOrProvider must be given");
+  if (!config.protocol[input.network]?.protocol.routerHlpBalancer) {
+    throw new Error(`routerHlpBalancer does not exist on ${input.network}`);
+  }
+
+  const path = getHlpToBalancerPath(input.fromToken, input.toToken);
+
+  const hlpQuote = await hlpSwap.quote({
+    ...input,
+    toToken: path.hlpBalancerToken
+  });
+
+  let balancerInAmount = BigNumber.from(hlpQuote.buyAmount);
+  if (path.tokenIn.address.toLowerCase() !== path.tokenOut.address.toLowerCase()) {
+    balancerInAmount = balancerInAmount
+      .mul(BASIS_POINTS_DIVISOR - hlpQuote.feeBasisPoints)
+      .div(BASIS_POINTS_DIVISOR);
+  }
+
+  // this is treated as the buy amount with no fee. In actuality, balancer pool does
+  // have fees, however they are calculated in a non-trivial way. It is identical to the
+  // user to not add the balancer fee to the quote, and only include the expected amount out
+  const expectedOut = await getWeightedPoolAmountOut(
+    balancerInAmount,
+    path.pool.poolId,
+    input.network,
+    path.hlpBalancerToken.address,
+    input.toToken.address
+  );
+
+  return {
+    allowanceTarget: config.protocol[input.network]!.protocol.routerHlpBalancer,
+    buyAmount: expectedOut.toString(),
+    feeBasisPoints: hlpQuote.feeBasisPoints,
+    feeChargedBeforeConvert: false,
+    gas: config.convert.gasEstimates.hlpToBalancer,
+    sellAmount: input.sellAmount.toString()
+  };
+};
+
+const hlpToBalancerTransaction = async (
+  input: ConvertTransactionRouteArgs
+): Promise<ethers.PopulatedTransaction> => {
+  const path = getHlpToBalancerPath(input.fromToken, input.toToken);
+  const routerAddress = mustExist(
+    config.protocol[input.network]?.protocol.routerHlpBalancer,
+    `routerHlpBalancer address on ${input.network}`
+  );
+
+  const router = RouterHlpBalancer__factory.connect(routerAddress, input.signer);
+
+  const pairs: Pair[] = [
+    {
+      baseSymbol: path.tokenIn.symbol,
+      quoteSymbol: "USD"
+    },
+    {
+      baseSymbol: path.hlpBalancerToken.symbol,
+      quoteSymbol: "USD"
+    }
+  ];

```

Comment:
`pairFromString` would be more concise here:
```suggestion
  const pairs: Pair[] = [
    pairFromString(`${path.tokenIn.symbol}/USD`),
    pairFromString(`${path.hlpBalancerToken.symbol}/USD`)
  ];

```

Reply:
I don't see the benefit of casting the pair to a string only to call pairFromString

Reply:
It's for readability, less code. We are not casting it to a string, it is already a string, we just concatenate it.